src/test/java/uk/gov/dwp/dataworks/provider/kms/KMSDataKeyGeneratorProviderTest.java

```diff
@@ -49,41 +50,41 @@ public void generateDataKey() {
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesNotFoundException() {
+    public void handlesNotFoundException() throws MasterKeystoreException {
         throwException(NotFoundException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesDisabledException() {
+    public void handlesDisabledException() throws MasterKeystoreException {
         throwException(DisabledException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesDependencyTimeoutException() {
+    public void handlesDependencyTimeoutException() throws MasterKeystoreException {
         throwException(DependencyTimeoutException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesInvalidKeyUsageException() {
+    public void handlesInvalidKeyUsageException() throws MasterKeystoreException {
         throwException(InvalidKeyUsageException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesInvalidGrantTokenException() {
+    public void handlesInvalidGrantTokenException() throws MasterKeystoreException {
         throwException(InvalidGrantTokenException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesKMSInternalException() {
+    public void handlesKMSInternalException() throws MasterKeystoreException {
         throwException(KMSInternalException.class);
     }
 
     @Test(expected = DataKeyGenerationException.class)
-    public void handlesKMSInvalidStateException() {
+    public void handlesKMSInvalidStateException() throws MasterKeystoreException {
         throwException(KMSInvalidStateException.class);
     }
 
-    private void throwException(Class<? extends Exception> e) throws CurrentKeyIdException {
+    private void throwException(Class<? extends Exception> e) throws CurrentKeyIdException, MasterKeystoreException {
```

Comment:
Is this testing the various catch blocks and their content?

Reply:
That's just a utility method that throws an exception. It's not new though and has not been introduced as part of this work - the signature had to be altered to accommodate a new exception.